components/gitpod-protocol/src/workspace-cluster.ts

```diff
@@ -104,6 +104,7 @@ export interface WorkspaceClusterDB {
      */
     findFiltered(predicate: DeepPartial<WorkspaceClusterFilter>): Promise<WorkspaceClusterWoTLS[]>;
 }
-export interface WorkspaceClusterFilter extends Pick<WorkspaceCluster, "state" | "govern" | "url"> {
+export interface WorkspaceClusterFilter
```

Comment:
To make `findFiltered` safer, we should make `applicationCluster` mandatory:
```
    findFiltered(predicate: WorkspaceClusterFilter): Promise<WorkspaceClusterWoTLS[]>;
...
export type WorkspaceClusterFilter = Pick<WorkspaceCluster, "applicationCluster"> & DeepPartial<Pick<WorkspaceCluster, "name" | "state" | "govern" | "url"> & { minScore: number }>
```
(or similar)

Reply:
made manadatory in 59ad36ec515b78410261e1b9422ec09c4b7ff73e